### DISCUSSION TITLE
QUESTION: How to find machines closest by address?

### DISCUSSION THREAD

**Discussion by dbrian57** [pinballmap/pbm#1374] [Q&A] [answered]
Hi! This API is super cool! 

I'm in a software dev class and a group of us are trying to integrate this API into our final project. We just wanted to know though if there was a way to return locations closest to an address. We see the `/api/v1/locations/closest_by_address.json` endpoint but that only returns one location. Is there a way to return all the locations available within a radius?

Thanks for any help on this!

**Comment by RyanTG** [comment, marked as answer]
Yes. `send_all_within_distance=true`. Example: 

```https://pinballmap.com/api/v1/locations/closest_by_address.json?send_all_within_distance=true;no_details=1;by_machine_id=2771;address=90065;max_distance=50```

**Reply by dbrian57** [reply]
You rock! Thank you so much.

**Reply by RyanTG** [reply]
Sweet! Feel free to share your final project with us!
